modules/stage-2.nix

```diff
@@ -4,8 +4,20 @@
 #        (This'll allow complex schemes like LVM)
 
 let
-  rootfs = config.fileSystems."/".device;
+  rootfs = config.fileSystems."/";
+  inherit (builtins) concatStringsSep elem length;
   inherit (lib) mkMerge mkOrder;
+
+  waitForRootfs =
+    lib.optionalString (elem "loop" rootfs.options) ''
```

Comment:
Should this maybe happen unconditionally? I think it would make sense for block devices carrying the root filesystem too.

Reply:
I think regular devices are better covered by `udevadm-settle` (which currently doesn’t work for unknown reasons).

Reply:
Oh, wait, yeah, I see what you are talking about. Yes, it’s probably not a bad idea to wait unconditionally.

Reply:
Although with "real" block devices I guess we could use udev event like I also suggested in https://github.com/samueldr/mobile-nixos/pull/35#issuecomment-547006432